catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -41,14 +46,25 @@
 
     private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
 
+    private static final String KARAF_LOCAL_ROLE = "karaf.local.roles";
+
     private final RegistryPublicationService registryPublicationService;
 
     private final ExecutorService executor;
 
+    private javax.security.auth.Subject subject;
+
     public RegistryPublicationHandler(RegistryPublicationService registryPublicationService,
             ExecutorService service) {
         this.registryPublicationService = registryPublicationService;
         this.executor = service;
+
+        Set<Principal> principals = new HashSet<>();
```

Comment:
Instead of copying and pasting this stuff all over the place. Can we just fix the Security class? You just need to replace it with the code that you've already got here. That will then fix everywhere else that is just using that class and you don't need all this duplicate code.